### PR TITLE
add immovable rod event

### PR DESCRIPTION
**Pull request by Slava0135**
## About the PR

adds new dangerous event and removes immovable rod launcher because it didn't work in the first place

**Media**

Peek 2023-06-12 13-33

**Changelog**

:cl:
- add: added rare immovable rod event that can happen late in the game and can literally cut through entire station
- fix: immovable rods now move on spawn by default
- tweak: you can now warp to immovable rod as ghost

### CODE REVIEW

**Comment by VigersRay** [comment]
I think it's funny but too illogical, if you want to cut the station in half with meaning, I think it's worth a look here.
https://github.com/space-wizards/space-station-14/assets/60344369/22894b17-f7fa-463b-ae4c-ea4edb631221
Of course it can be done faster.

**Comment by GoodWheatley** [comment]
The rod literally cutting the station in half is effectively an instant round-end since there’s no way to put the station back together. If the event could use a variant of the rod that didn’t destroy floors, I think it would work much better as a midround event.

**Comment by VigersRay** [comment]
> The rod literally cutting the station in half is effectively an instant round-end since there’s no way to put the station back together. If the event could use a variant of the rod that didn’t destroy floors, I think it would work much better as a midround event.

Even so, if the station is as big as a box, it will kill half the crew and there will be no opportunity to restore the station

**Comment by Slava0135** [comment]
Right now it can spawn 1 hour and 30 minutes into the game and has same chance to happen as space dragon. Plus trajectory is random it can potentially only "cut corners", so station cut in half should be not that common overall.

**Comment by Slava0135** [comment]
> > Right now it can spawn 1 hour and 30 minutes into the game and has same chance to happen as space dragon. Plus trajectory is random it can potentially only "cut corners", so station cut in half should be not that common overall.
> 
> My opinion on how it should work is:
> 
>     * Rod can’t destroy floors
> 
>     * Can happen 60 minutes in, same chance as lone op
> 
> 
> There isn’t really any random event that engineering can interact with right now, and a tame version of the immovable rod that didn’t split the station into two pieces would fill that niche by creating a repair job for engineers.

there are meteors

**Comment by Just-a-Unity-Dev** [comment]
also "high velocity" doesn't make sense

the universe is moving around it not the other way around :godo:

**Comment by Slava0135** [comment]
> rod MUST be an admin event, it should never spawn normally
> 
> > remove: removed immovable rod launcher
> 
> why

because it didn't work, rod flies like 3 tiles max and stops

**Comment by Just-a-Unity-Dev** [comment]
> > also "high velocity" doesn't make sense
> > the universe is moving around it not the other way around :godo:
> 
> its all relative

its called the immovable rod

**Comment by EmoGarbage404** [comment]
this is honestly funny maybe make it have an even lower weight but i love this.

**Comment by deltanedas** [comment]
with pro spacing you have a solid 5 minutes to fix the breaches before its dangerous, low hazard event

**Comment by Slava0135** [comment]
i made it not destroy tiles (fun fact - it can destroy gravity generator, have fun flying!) and appear earlier into shift (45 min)

**Comment by Slava0135** [comment]
fun fact - tg station has it and i only discovered it now
image

File: Content.Server/ImmovableRod/ImmovableRodSystem.cs [hunk 1]
```diff
@@ -44,31 +45,22 @@ public override void Initialize()
         base.Initialize();
 
         SubscribeLocalEvent<ImmovableRodComponent, StartCollideEvent>(OnCollide);
-        SubscribeLocalEvent<ImmovableRodComponent, ComponentInit>(OnComponentInit);
```
Comment (AJCM-git) [inline Content.Server/ImmovableRod/ImmovableRodSystem.cs:47]: it should keep the old behaviour when spawned

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml [hunk 2]
```diff
@@ -356,16 +356,3 @@
       tags:
         - CartridgeRocket
     proto: MeteorLarge
-
-- type: entity
-  name: immovable rod launcher
-  parent: WeaponLauncherAdmemeMeteorLarge
-  id: WeaponLauncherAdmemeImmovableRodSlow
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml:363]: eh, its not player-facing and someone will fix it someday

**Review by AJCM-git — CHANGES_REQUESTED** [review]
I think the reviews can be resumed as "keep the old rod behaviours and fields, all the event should do is store a prototype id of the rod and change its direction towards the station"

**Comment by metalgearsloth** [comment]
Kinda suss because every other event is recoverable besides this one.

**Comment by Emisse** [comment]
maybe make a variant of it that doesnt destroy tiles for the event version

**Comment by Slava0135** [comment]
I made it not destroy floors

**Comment by Emisse** [comment]
ok its good then

**Comment by deltanedas** [comment]
i think What the fuck was that? would be better announcement

File: Content.Server/ImmovableRod/ImmovableRodSystem.cs (hunk 1 above)
Comment (Slava0135) [reply Content.Server/ImmovableRod/ImmovableRodSystem.cs:47]: actually it keeps old behaviour - rod does not move when spawned

Comment (Slava0135) [reply Content.Server/ImmovableRod/ImmovableRodSystem.cs:47]: like, rod spawn code is pretty cursed so I got rid of it

**Comment by Slava0135** [comment]
Peek 2023-07-01 16-17
rods now move on spawn (i guess it was a bug because there is `RandomizeVelocity` field set to true by default)

**Comment by Slava0135** [comment]
rod launcher still doesn't work properly

File: Content.Server/StationEvents/Components/ImmovableRodRuleComponent.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+using Content.Server.StationEvents.Events;
+
+namespace Content.Server.StationEvents.Components;
+
+[RegisterComponent, Access(typeof(ImmovableRodRule))]
+public sealed class ImmovableRodRuleComponent : Component
+{
+    [DataField("rodPrototype")]
```
Comment (AJCM-git) [inline Content.Server/StationEvents/Components/ImmovableRodRuleComponent.cs:8]: prototypeidserializer

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**